benchmarking/blackbox_repository/blackbox_offline.py

```diff
@@ -82,18 +82,29 @@ def _objective_function(
         # todo: we should check range configuration with configspaces
         # query the configuration in the list of available ones
         key_dict = configuration
-        if self.fidelity_space is not None:
-            key_dict.update(fidelity)
         if self.seed_col is not None:
             key_dict[self.seed_col] = seed
-        key = tuple(key_dict[col] for col in self.index_cols)
-        try:
-            return dict(self.df.loc[key, self.metric_cols])
-        except KeyError:
+        if fidelity is not None:
+            if self.fidelity_space is not None:
+                key_dict.update(fidelity)
+            else:
+                raise ValueError("fidelity is not None and self.fidelity_space is None.")
+        if fidelity is None and self.fidelity_space is not None:
+            keys = tuple(set(self.index_cols) - set(self.fidelity_space.keys()))
+        else:
+            keys = self.index_cols
+        output = self.df.xs(tuple(key_dict[col] for col in keys), level=keys).loc[:, self.metric_cols]
+        if len(output) == 0:
             raise ValueError(
                 f"the hyperparameter {configuration} is not present in available evaluations. Use `add_surrogate(blackbox)` if"
                 f" you want to add interpolation or a surrogate model that support querying any configuration."
             )
+        if fidelity is not None:
+            return output.iloc[0].to_dict()
+        else:
+            # TODO select only the fidelity values in the self.fidelity_space, since it might be the case there are more
+            #  values in the dataframe. Then the output tensor has larger number of elements than expected num_fidelities.
+            return output
```

Comment:
At the moment, for `fidelity=None` it's returning a Pandas DataFrame with shape (num_fidelities, num_objectives) rather than a numpy tensor. Should we change it to output a tensor?

Reply:
Changed to output a numpy tensor.

Reply:
Thanks, makes sense.